### ISSUE TITLE
Transfer-encoding is set to chunked even if content-length is known

### ISSUE THREAD

**Issue by adamw** [zio/zio-http#679]
The following code:

```scala
import zio._
import zhttp.http._
import zhttp.service.Server
import zio.stream.ZStream

object Test extends App {
  val app = Http.collect[Request] { case _ =>
    Response(
      status = Status.OK,
      headers = List(Header.contentLength(4)),
      data = HttpData.fromStream(ZStream.fromIterable("test".getBytes))
    )
  }

  override def run(args: List[String]): URIO[zio.ZEnv, ExitCode] =
    Server.start(8090, app).exitCode
}
```

returns the following response:

```
$ curl --raw -v http://localhost:8090

< HTTP/1.1 200 OK
< content-length: 4
< transfer-encoding: chunked
<
4
test
0
```

Which is incorrect as the content is longer (due to the chunks delimiters). If the length is known, I think a normal response should be returned.

I'm using version `1.0.0.0-RC19`.

**Comment by tusharmath** [comment]
This is a bug — 

More info available here — https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Transfer-Encoding#directives

> Data is sent in a series of chunks. The Content-Length header is omitted in this case and at the beginning of each chunk you need to add the length of the current chunk in hexadecimal format, followed by '\r\n' and then the chunk itself, followed by another '\r\n'. The terminating chunk is a regular chunk, with the exception that its length is zero. It is followed by the trailer, which consists of a (possibly empty) sequence of header fields.

**Comment by tusharmath** [comment]
For this issue I think we can just drop the `transfer-encoding` completely from Response creation.

**Comment by adamw** [comment]
Well I suspect it depends ... if the content length is known, then yes you can send a normal response. But if the length is unknown, then I suppose you should send chunks?

**Comment by tusharmath** [comment]
Yes, you can use chunked encoding or you can send the file/stream and close the connection once the stream is over. What do you think?
Chunked encoding can be added if the user applies a special operator, eg:

```scala
response.withChunkedEncoding // adds transfer-encoding header and actually does chunked encoding.
```

OR 

We can use chunked encoding if the header contains `transfer-encoding: chunked` header.

It is also possible to implement chunked encoding in user-land code using ZStreams, which can then get quite messy :D

**Comment by adamw** [comment]
Another possibility: `HttpData.fromChunkStream(ZStream[R, E, Chunk[Byte]])`. I know there are chunks behind the stream so in fact that's double-chunking, but the user has then full control over the chunk boundaries. But maybe that's too low-level :)

**Comment by tusharmath** [comment]
Fixed in #711

**Comment by adamw** [comment]
thanks, works :)
